product-service/handlers/getProductById.js

```diff
@@ -8,21 +8,22 @@ const cors = {
 const buildResponse = (statusCode, body, headers) => ({statusCode, body: JSON.stringify(body), headers});
 
 const handler = async (event) => {
+  console.log('Incoming event: ', event);
   const id = event.pathParameters?.id;
 
   try {
-    const products = await getMockedProducts();
+    const { rows: products } = await query(`SELECT p.id, p.title, p.description, p.price, p.image, s.count FROM product p INNER JOIN stock s ON p.id = s.product_id WHERE p.id::text = $1`, [id]);
 
     if (!products) {
       throw new Error(JSON.stringify(buildResponse(500, {message:`Failed to retrieve products`})));
     }
 
-    const product = products.find(p => p.id === id);
+    const product = products[0];
     if (!product) {
       throw new Error(JSON.stringify(buildResponse(404, {message:`Failed to retrieve product with id ${id}`})));
     }
 
-    return buildResponse(200, product, cors);
+    return buildResponse(200, products, cors);
```

Comment:
Shouldn't you return an object instead? It will be strange to get an array on one object request, isn't it?

Reply:
The good point of view. I will keep this in mind in the future.